src/features/inlay_hints.zig

```diff
@@ -69,23 +69,22 @@ const Builder = struct {
 
         var converted_hints = try self.arena.alloc(types.InlayHint, self.hints.items.len);
         for (converted_hints, self.hints.items) |*converted_hint, hint| {
-            const index = offsets.tokenToIndex(self.handle.tree, hint.token_index);
             const position = offsets.advancePosition(
                 self.handle.tree.source,
                 last_position,
                 last_index,
-                index,
+                hint.index,
                 offset_encoding,
             );
-            defer last_index = index;
+            defer last_index = hint.index;
             defer last_position = position;
             converted_hint.* = types.InlayHint{
                 .position = position,
                 .label = .{ .string = hint.label },
                 .kind = hint.kind,
-                .tooltip = .{ .MarkupContent = hint.tooltip },
+                .tooltip = if (hint.tooltip != null) .{ .MarkupContent = hint.tooltip.? } else null,
```

Comment:
You can do `if (some_optional) |inner_value| { ... } else { ... }` to conditionally capture the value inside an optional so that you can avoid the `.?` which is slightly less idiomatic.
This is how I would do it:
```suggestion
                .tooltip = if (hint.tooltip) |tooltip| .{ .MarkupContent = tooltip } else null,
```
I don't expect you to do this change. I just wanted you to inform you about this since you said that you are new to zig and you may not have known about this language feature.

Reply:
Thanks!